A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,554 @@
+      // Maybe ok to support a custom binary propagator. Needs based64 encoding 
+      // validation if so. Not for now.
+      gpr_log(GPR_ERROR, "Binary propagator other than GrpcTraceBinPropagator is not supported.");
+      return "";
+    }
+    return metadata_->GetStringValue(key);
+  }
+
+  void Set(nostd::string_view key, nostd::string_view value) {
+    if (key == "grpc-trace-bin") {
+      metadata_->Set(
+          grpc_core::GrpcTraceBinMetadata(),
+          grpc_core::Slice::FromCopiedString(absl::Base64Unescape(value)));
+    } else if (absl::EndsWith(key, "-bin")) {
+      gpr_log(GPR_ERROR, "Binary propagator other than GrpcTraceBinPropagator is not supported.");
+      return;
+    } else {
+      // A propagator other than GrpcTraceBinTextMapPropagator was used.
+      metadata_->Append(key, grpc_core::Slice::FromCopiedString(value));
+    }
+  }
+
+ private:
+  grpc_metadata_batch* metadata_;
+};
+
+}  // namespace internal
+
+std::unique_ptr<opentelemetry::context::TextMapCarrier>
+MakeGrpcTraceBinTextMapPropagator() {
+  return std::make_unique<internal::GrpcTraceBinTextMapPropagator>();
+}
+
+}  // namespace grpc
+
+```
+
+### gRPC OpenTelemetry Tracing API
+This section talks about enabling and configuring OpenTelemetry tracing.
+The OpenTelemetry API will coexist with the OpenCensus API. Only one 
+`grpc-trace-bin` header will be sent for a single RPC as long as only one of 
+OpenTelemetry or OpenCensus is enabled for the channel.
+
+The APIs are different among languages due to different underlying infrastructures.
+
+#### Java
+In Java, it will be part of global interceptors, so that the interceptors are 
+managed in a more sustainable way and user-friendly. As a prerequisite, the stream
+tracer factory API will be stabilized. OpenTelemetryModule will be created with 
+an OpenTelemetryAPI instance passing in for necessary configurations. 
+Users can also rely on SDK autoconfig extension that configures the sdk object 
+through environmental variables or Java system properties, then obtain the sdk 
+object passed in to gRPC.
+
+```Java
+// Construct OpenTelemetry to be passed to gRPC OpenTelemetry module for 
+// trace and metrics configurations.
+SdkTracerProvider sdkTracerProvider = SdkTracerProvider
+  .builder()
+  .addSpanProcessor(
+     BatchSpanProcessor.builder(exporter).build())
+  .build();
+
+OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+  .setTracerProvider(sdkTracerProvider)
+  .setMeterProvider(...)
+  .setPropagators(
+     ContextPropagators.create(GrpcTraceBinPropagator.getInstance()))
+  .build();
+
+// Alternatively, use auto configuration:
+// OpenTelemetry openTelemetry = 
+// AutoConfiguredOpenTelemetrySdk.getOpenTelemetrySdk().
+
+// Create a module that hosts tracing infrastructures. Should document that 
+// the module implementation may change as OpenTelemetry evolves.
+OpenTelemetryModule otModule = OpenTelemetryModule.getInstance(openTelemetry);
```

Comment:
Why is this not using `OpenTelemetryPluginBuilder` as in other languages?  I think we want that to be the main API for all things we configure via OT, in all languages.  Tracing should not have a fundamentally different API from metrics.

Reply:
Just a naming issue. The APIs are the same: tracing and metrics share the same plugin. Updated.

Reply:
Java has `OpenTelemetryModule` equivalent to `OpenTelemetryPluginBuilder`. Align with A66. Let me know if you have concerns on the name.